Formula/node.rb

```diff
@@ -56,30 +56,11 @@ def install
 
     if build.with? "npm"
       resource("npm").stage buildpath/"npm_install"
-
-      # make sure npm can find node
-      ENV.prepend_path "PATH", bin
```

Comment:
IMHO a breakage in an unsupported functionality, which was never intented to work, is not a breaking change which must be explicitly mentioned in the breaking changes section. Nevertheless npm@5 is a semver major release, so breaking changes are expected anyway. 

Regarding yarn + node-gyp: It's not about where it is located in libexec, its about where is it linked too. [Yarn will only look](https://github.com/yarnpkg/yarn/blob/8ede3c41cfb389ee408ae1094503649dc11b11cc/src/util/execute-lifecycle-script.js#L133-L142) in the global bin folder `/user/local/lib/node_modules/npm/bin/node-gyp-bin/node-gyp` (which worked previously) and the 2 path relative to bin/node which does not work fore homebrew, because npm is inside libexec and not lib.